packages/swing-store/src/snapStore.js

```diff
@@ -154,72 +140,30 @@ export function makeSnapStore(
           snapReader.destroy();
         });
 
-        // Create a file for the compressed snapshot in-place
-        // to be atomically renamed once we know its name
-        // from the hash of raw snapshot contents.
-        const {
-          path: tmpGzPath,
-          fd: tmpGzFd,
-          cleanup: tmpGzCleanup,
-        } = await ptmpFile({ tmpdir: root });
-        cleanup.push(tmpGzCleanup);
-        const gzWriter = createWriteStream(noPath, {
-          fd: tmpGzFd,
-          autoClose: false,
-        });
-        cleanup.push(() => gzWriter.close());
-
-        await Promise.all([fsStreamReady(snapReader), fsStreamReady(gzWriter)]);
+        await fsStreamReady(snapReader);
 
         const hashStream = createHash('sha256');
         const gzip = createGzip();
+        let compressedByteCount = 0;
 
         const { result: hash, duration: compressSeconds } =
           await measureSeconds(async () => {
             snapReader.pipe(hashStream);
-            snapReader.pipe(gzip).pipe(gzWriter);
-
-            await Promise.all([finished(snapReader), finished(gzWriter)]);
+            const snapshot = await buffer(snapReader.pipe(gzip));
```

Comment:
It's great that Node.js has a built-in for this now! I have the following helper to deal with this situation for earlier Node versions:

```js
/**
 * @param {AsyncIterable<Buffer>} res
 */
export const asBuffer = async (res) => {
  const chunks = [];
  for await (const chunk of res) {
    chunks.push(chunk);
  }

  return Buffer.concat(chunks);
};
```

used as
```suggestion
            const snapshot = await asBuffer(snapReader.pipe(gzip));
```